custom-plugin-rules/markdown-validator/README.md

```diff
@@ -0,0 +1,188 @@
+
+```json
+{
+  "name": "redocly-openapi-markdown-plugin",
+  "version": "1.0.0",
+  "description": "",
+  "main": "openapi-markdown.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "markdownlint": "^0.31.1"
+  }
+}
+```
+
+Make sure to install the dependency using your favorite package manager. For example I use `npm` so my installation command is:
+
+```
+npm install
+```
+
+The entry point for the plugin code is in `openapi-markdown.js`:
+
+```js
+const ValidateMarkdown = require('./rule-validate-markdown.js');
+
+module.exports = {
+  id: 'openapi-markdown',
+  rules: {
+    oas3: {
+      'validate': ValidateMarkdown,
+    }
+  }
+	
+}
+```
+
+The rule itself is in `rule-validate-markdown.js`:
+
+```js
+module.exports = ValidateMarkdown
+const markdownlint = require("markdownlint");
+const config = {
+  // the list is here https://github.com/DavidAnson/markdownlint#rules--aliases
+  MD013: {line_length: 120},
+  MD041: false, // first line should be h1
+  MD047: false, // should end with newline
+}
+
+function checkString(description, ctx) {
+  let options = {
+    "strings": {
+      "desc": description
+    },
+    "config": config
+  };
+  markdownlint(options, function callback(err, result) {
+    if (!err) {
+      // if there's no problem do nothing
+      if (result.desc.length) { // desc is the key in the options.strings object
+        let lines = description.split("\n");
+
+        result.desc.forEach((desc) => {
+          message = desc.ruleDescription;
+          // add line number context for longer entries
+          if (desc.lineNumber > 1 ) {
+            message = message + " (near: " + lines[desc.lineNumber].substring(0,20) + "... )";
+          }
+
+          ctx.report({
+            message: message,
+            location: ctx.location.child('description'),
+          });
+        });
+
+      }
+    }
+  });
+}
+
+function ValidateMarkdown() {
+  console.log("OpenAPI Markdown: validate");
+  return {
+   Info: {
+      enter(target, ctx) {
+        if(target["description"]) {
+          return checkString(target["description"], ctx);
+          
+        }
+      }
+    },
+   Operation: {
+      enter(target, ctx) {
+        if(target["description"]) {
+          return checkString(target["description"], ctx);
+          
+        }
+      }
+    },
+   Parameter: {
+      enter(target, ctx) {
+        if(target["description"]) {
+          return checkString(target["description"], ctx);
+          
+        }
+      }
+    },
+    Tag: {
+      enter(target, ctx) {
+        if(target["description"]) {
+          return checkString(target["description"], ctx);
+          
+        }
+      }
+    },
+  }
+}
+```
+
+To control the markdown validation rules in use, edit the config at the top of the file. 
+
+Bring the plugin into your `redocly.yaml` file like this:
+
+```yaml
```

Comment:
Q -- thoughts on including a local openapi for the linter to run against and adding the api to this config file? 

I had to go look up the syntax in the docs to add an api when I was testing.

Reply:
I thought about this when we started the cookbook but I think it adds more complexity than the value it provides. The target audience here is people who are already Redocly CLI users - so I'm assuming they have both an OpenAPI file and a problem to solve without us supplying either of those!